index.js

```diff
@@ -0,0 +1,448 @@
+            let div = document.createElement('div');
+            div.className = 'key';
+            div.setAttribute('data-key', prop);
+            div.classList.add(prop);
+            if (Array.isArray(this.keysValue[prop])) {
+                if (prop.indexOf('Digit') === -1 && prop !== 'Backquote' && prop !== 'Minus' && prop !== 'Equal') {
+                    div.innerHTML = `${this.keysValue[prop][this.langSwitcher]}`;
+                }
+                else
+                    div.innerHTML = `${this.keysValue[prop][this.shiftSwitcher]}`;
+            }
+            else {
+                div.innerHTML = `${this.keysValue[prop]}`;
+            }
+            if (div.innerHTML.length > 1) {
+                div.setAttribute('data-special', 'specialKey');
+            }
+            if (prop.includes('Shift') || prop.includes('Tab') || prop.includes('Caps') || prop.includes('Enter') ||
+                prop.includes('Backspace')) {
+                div.classList.add('bigKey');
+            }
+            if (prop.includes('Space')) {
+                div.classList.add('keySpace');
+            }
+            keyRaw.append(div);
+
+            if (prop === 'Backspace') {
+                keyRaw.classList.add(`raw-${++counter}`);
+                container.append(keyRaw);
+                keyRaw = document.createElement('div');
+            }
+            if (prop === 'Backslash') {
+                container.append(keyRaw);
+                keyRaw.className = 'raw flex';
+                keyRaw.classList.add(`raw-${++counter}`);
+                keyRaw = document.createElement('div');
+            }
+            if (prop === 'Enter') {
+                container.append(keyRaw);
+                keyRaw.className = 'raw flex';
+                keyRaw.classList.add(`raw-${++counter}`);
+                keyRaw = document.createElement('div');
+            }
+            if (prop === 'ShiftRight') {
+                container.append(keyRaw);
+                keyRaw.className = 'raw flex';
+                keyRaw.classList.add(`raw-${++counter}`);
+                keyRaw = document.createElement('div');
+            }
+            if (prop === 'Delete') {
+                container.append(keyRaw);
+                keyRaw.className = 'raw flex';
+                keyRaw.classList.add(`raw-${++counter}`);
+            }
+        }
+        let textarea = document.createElement('textarea');
+        textarea.className = 'textarea';
+        textarea.setAttribute('type', 'text');
+        textarea.id = 'textarea';
+        container.prepend(textarea);
+        fragment.append(container);
+        document.body.append(fragment);
+    };
+}
+
+let changeKeysLang = keyEventFunctions.hotKeysAction;
+let copy = keyEventFunctions.hotKeysAction;
+let paste = keyEventFunctions.hotKeysAction;
+let cut = keyEventFunctions.hotKeysAction;
+
+
+window.onload = function () {
+    let keyBoard = new Keyboard();
```

Comment:
Architecture notice but I'll try to explain here.

Good part:
3 objects
1) Keyboard (View)
2) keyEventFunctions (logic of working with textArea)
3) mainInputFunction (logic of handling inputs from user)

Bad part:
High coupling between all of them (2 bound to 1 using this `this.keysValue`, 3 supports this by using `.call(context)`)

Possible solution:
Composition concept (https://en.wikipedia.org/wiki/Composition_over_inheritance)
```javascript
const keyBoard = new Keyboard();
const keyEvents = new KeyEvents();
keyBoard.setKeys(keyEvents); 

const mainInput = new MainInput(keyBoard);
```

It's not gonna work like this but I wanted to show that by this approach we can easily (on run-time) change keys logic implementation from our **View** and **User handler** layers. 

Actually you did almost the same but by using the global variable `context` and `.call` method and the result we're losing some flexibility with that. So your thoughts about architecture were right but composition over `.call()` and global variables!